### ISSUE TITLE
Nodes can't return bytes objects. Bonobo incorrectly iterates over it.

### ISSUE THREAD

**Issue by TZubiri** [python-bonobo/bonobo#186]
The following code should print "test" once:

```

import bonobo

def extract():
    return b'test'

def load(bytes):
    print(bytes.decode('UTF-8'))

graph = bonobo.Graph(extract,load)

bonobo.run(graph)
```

Instead it produces the following error message 4 times (one for each character in the bytes object):
```
AttributeError (in function)
 | 'int' object has no attribute 'decode'
 | 
 | Traceback (most recent call last):
 |   File "C:\Python36\lib\site-packages\bonobo-0.4.3-py3.6.egg\bonobo\execution\node.py", line 88, in loop
 |     self.step()
 |   File "C:\Python36\lib\site-packages\bonobo-0.4.3-py3.6.egg\bonobo\execution\node.py", line 111, in step
 |     self.handle_results(input_bag, input_bag.apply(self._stack))
 |   File "C:\Python36\lib\site-packages\bonobo-0.4.3-py3.6.egg\bonobo\structs\bags.py", line 63, in apply
 |     return func_or_iter(*args, *self.args, **kwargs, **self.kwargs)
 |   File "C:\Python36\lib\site-packages\bonobo-0.4.3-py3.6.egg\bonobo\config\processors.py", line 85, in __call__
 |     return self.wrapped(*self.context, *args, **kwargs)
 |   File "C:/SVN3/Harvester/Parsing/PC/iShares/test/testParallelDownload.py", line 38, in load
 |     print(bytes.decode('UTF-8'))
 | AttributeError: 'int' object has no attribute 'decode'

```

When we change the load function to:

`
def load(bytes):
    print(bytes)
`

The following is printed

116
101
115
116

Which means that bonobo is iterating over the bytes object and printing each character as an int.

**Comment by TZubiri** [comment]
I'm sure there are better alternatives. But an idea that is really good regardless of the solution to this problem, would be utilizing type hints as part of the Graph api. For this specific case, we would be using them to disambiguate between 2 interpretations.

We would have the option to define the function as follows

```
def extract()->bytes:
def load(bytes:bytes): 
```
which would result in 'test' being printed.

or as follows:
``
def extract()->int:

def load(int):
``

Which would result in 116 101 115 116 being printed.

**Comment by hartym** [comment]
I think it's safe to say that if it does not consider a string as something to be iterated on, it should do the same for bytes. Question is what is the dichotomy between iterables that should be iterated on, or not ... Maybe the answer should be "never iterated on return values, force yield to be used if it is what you want."

**Comment by cw-andrews** [comment]
I personally really like that solution Romain. Definitely a very simple and
intuitive solution!

**Comment by hartym** [comment]
Just commited a fast and dirty fix, but will probably go for the more explicit way described above in next minor.

**Comment by hartym** [comment]
Released in 0.5.1
